### PR TITLE
Add support for the Eta template engine

### PR DESCRIPTION
**Pull request by nebrelbug**
This PR adds support for the [Eta](https://eta.js.org) template engine (see #178).

Since Eta and EJS have similar APIs, I copied most of the tests and templates from the EJS ones. 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
  - I added Eta to the benchmarks folder just to gauge performance, but am guessing I should remove it?
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/test-eta.js [hunk 1]
```diff
@@ -0,0 +1,843 @@
...
+  collapseWhitespace: true,
+  collapseBooleanAttributes: true,
+  removeAttributeQuotes: true,
+  removeEmptyAttributes: true
+}
+
+test('reply.view with eta engine and custom templates folder', t => {
+  t.plan(6)
+  const fastify = Fastify()
+  const eta = require('eta')
```
Comment (mcollina) [inline test/test-eta.js:22]: move all the `require('eta')` to the top.

File: test/test-eta.js [hunk 2]
```diff
@@ -0,0 +1,843 @@
...
+  const data = { text: 'text' }
+
+  fastify.register(require('../index'), {
```
Comment (mcollina) [inline test/test-eta.js:25]: also this require

File: test/test-eta.js [hunk 3]
```diff
@@ -0,0 +1,843 @@
...
+      eta: eta
+    },
+    includeViewExtension: true,
+    templates: templatesFolder,
+    options: options
+  })
+
+  fastify.get('/', (req, reply) => {
+    reply.type('text/html; charset=utf-8').view('index-linking-other-pages', data) // sample for specifying with type
+    // reply.view('index-with-includes', data)
```
Comment (mcollina) [inline test/test-eta.js:577]: why is this commented?

Comment (nebrelbug) [reply test/test-eta.js:577]: TBH, I'm not really sure (these tests are just modified from EJS'). My guess is that it was an earlier version of the test -- I'll remove it.

File: test/test-eta.js (hunk 2 above)
Comment (nebrelbug) [reply test/test-eta.js]: Just fixed it :smiley:

File: test/test-eta.js (hunk 1 above)
Comment (nebrelbug) [reply test/test-eta.js]: Just fixed it :smiley:

**Comment by nebrelbug** [comment]
I haven't added documentation about using Eta specifically, but am planning to add it to the Eta website.

**Review by mcollina — APPROVED** [review]
lgtm

File: index.js [hunk 4]
```diff
@@ -35,7 +35,7 @@ function fastifyView (fastify, opts, next) {
   const defaultCtx = opts.defaultContext || {}
   const layoutFileName = opts.layout
 
-  if (layoutFileName && type !== 'handlebars' && type !== 'ejs') {
+  if (layoutFileName && type !== 'handlebars' && type !== 'ejs' && type !== 'eta') {
```
Comment (zekth) [inline index.js:38]: Seems like the error message is not revelant anymore.

**Review by zekth — CHANGES_REQUESTED** [review]
Just a minor comment.

File: index.js (hunk 4 above)
Comment (nebrelbug) [reply index.js:38]: I think it's there because only Handlebars, EJS, and Eta support the `layout` option (support is added w/ the `withLayout` function).

Comment (zekth) [reply index.js:38]: Could you just edit the error message after this condition to be more relevant?

**Review by Ethan-Arrowood — APPROVED** [review]
this is well done! good work and thank you for contributing

File: index.js (hunk 4 above)
Comment (nebrelbug) [reply index.js:38]: @zekth just updated the error message, let me know what you think :)

**Review by zekth — APPROVED** [review]
LGTM

**Review by lilpoozie2005 — APPROVED** [review]

**Pull request merged**